src/matter/session/secure/CaseMessages.ts

```diff
@@ -15,56 +23,79 @@ export const KDFSR3_INFO = Buffer.from("Sigma3");
+
+/** @see {@link MatterCoreSpecificationV1_0} § 3.3 */
+const CRYPTO_HASH_LEN_BYTES = 32;
+const CRYPTO_HASH_BLOCK_LEN_BYTES = 64;
+
+/** @see {@link MatterCoreSpecificationV1_0} § 3.6 */
+const CRYPTO_SYMMETRIC_KEY_LENGTH_BITS = 128;
+const CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES = 16;
+const CRYPTO_AEAD_MIC_LENGTH_BITS = 128;
+const CRYPTO_AEAD_MIC_LENGTH_BYTES = 16;
+const CRYPTO_AEAD_NONCE_LENGTH_BYTES = 13;
+
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma1T = ObjectT({
-    random: Field(1, ByteStringT()),
-    sessionId: Field(2, UnsignedIntT),
-    destinationId: Field(3, ByteStringT()),
-    ecdhPublicKey: Field(4, ByteStringT()),
-    mrpParams: OptionalField(5, ObjectT({
+    random: Field(1, ByteStringT({ length: 32 })),
+    sessionId: Field(2, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
+    destinationId: Field(3, ByteStringT({ length: CRYPTO_HASH_LEN_BYTES })),
+    ecdhPublicKey: Field(4, ByteStringT({ length: CRYPTO_PUBLIC_KEY_SIZE_BYTES })),
+    mrpParams: OptionalField(5, ObjectT({ // sed-parameter-struct
         idleRetransTimeoutMs: OptionalField(1, UnsignedIntT),
         activeRetransTimeoutMs: OptionalField(2, UnsignedIntT),
     })),
-    resumptionId: OptionalField(6, ByteStringT()),
-    resumeMic: OptionalField(7, ByteStringT()),
+    resumptionId: OptionalField(6, ByteStringT({ length: 16 })),
+    resumeMic: OptionalField(7, ByteStringT({ length: CRYPTO_AEAD_MIC_LENGTH_BYTES})),
 });
 
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma2T = ObjectT({
-    random: Field(1, ByteStringT()),
-    sessionId: Field(2, UnsignedIntT),
-    ecdhPublicKey: Field(3, ByteStringT()),
-    encrypted: Field(4, ByteStringT({ maxLength: 400 })), // TODO: check max length in specs
-    mrpParams: OptionalField(5, ObjectT({
+    random: Field(1, ByteStringT({ length: 32 })),
+    sessionId: Field(2, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
+    ecdhPublicKey: Field(3, ByteStringT({ length: CRYPTO_PUBLIC_KEY_SIZE_BYTES })),
+    encrypted: Field(4, ByteStringT()),
+    mrpParams: OptionalField(5, ObjectT({ // sed-parameter-struct
         idleRetransTimeoutMs: OptionalField(1, UnsignedIntT),
         activeRetransTimeoutMs: OptionalField(2, UnsignedIntT),
     })),
 });
 
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma2ResumeT = ObjectT({
-    resumptionId: Field(1, ByteStringT()),
-    resumeMic: Field(2, ByteStringT()),
-    sessionId: Field(3, UnsignedIntT),
+    resumptionId: Field(1, ByteStringT({ length: 16 })),
+    resumeMic: Field(2, ByteStringT({ length: 16 })),
+    sessionId: Field(3, BoundedUnsignedIntT({ min: 0, max: 0xFFFF })),
 });
 
+/** @see {@link MatterCoreSpecificationV1_0} § 4.13.2.3 */
 export const CaseSigma3T = ObjectT({
-    encrypted: Field(1, ByteStringT({ maxLength: 400 })), // TODO: check max length in specs
+    encrypted: Field(1, ByteStringT()),
```

Comment:
This has an implicit maximal length somehow.
Encrypted data are: newOpCert | intermediateCACert | signature
So 400 + 400 + 64 + 16 (auth tag length) = 880

Maybe the best way to handle that is to create an Encrypted() method that will automatically create a ByteStringT with the right max.

Reply:
If you say that this is the limit ... I checked docs and were not able to explicitely find these number in the Case specs ... but maybe also just because I'm not that deep in there.

Will put the 880 as limit tomorrow

Extra type I would leave up to you ;-)

Reply:
I will try to do a PR now to add this method.

Reply:
maybe lets handle that really in follow up PR because for now it is proofed to work that way